op.c

```diff
@@ -16543,19 +16543,22 @@ Perl_subsignature_append_positional(pTHX_ OP *varop, OPCODE defmode, OP *defexpr
     if(defexpr) {
         signature->optelems++;
 
-        I32 flags = 0;
-        if(defmode == OP_DORASSIGN)
-            flags |= OPpARG_IF_UNDEF << 8;
-        if(defmode == OP_ORASSIGN)
-            flags |= OPpARG_IF_FALSE << 8;
-
         if(defexpr->op_type == OP_NULL && !(defexpr->op_flags & OPf_KIDS))
         {
+            /* newOP(OP_NULL, 0) in the caller */
```

Comment:
It took me a while to work out what this comment means. It was unclear if this is expecting to *cause* the caller to do that, or if this had been passed in as that. I think it would be clearer (if this is correct?) as:

```
/* Caller passed in newOP(OP_NULL, 0) here, so we should not leak it */
```